### PR TITLE
Cloud API optimizations

### PR DESCRIPTION
**Pull request by Aleffio**
**Description**
- Added phpdoc where missing
- Added Storeid
- Use array instead of jsondecode(jsonencode()) for the receipt

### CODE REVIEW

File: Gateway/Http/Client/TransactionPosCloudSync.php [hunk 1]
```diff
@@ -30,36 +30,41 @@
...
         \Adyen\Payment\Model\RecurringType $recurringType,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
         array $data = []
     ) {
         $this->_encryptor = $encryptor;
         $this->_adyenHelper = $adyenHelper;
         $this->_adyenLogger = $adyenLogger;
         $this->_recurringType = $recurringType;
         $this->_appState = $context->getAppState();
+        $this->storeManager = $storeManager;
```
Comment (cyattilakiss) [inline Gateway/Http/Client/TransactionPosCloudSync.php]: please create a property for the storemanager also

Comment (rikterbeek) [reply Gateway/Http/Client/TransactionPosCloudSync.php]: do we need to expose the storeManager. As it is only used for storeId we can just do: 
```
this->storeId = $storeManager->getStore()->getId();
```

File: Helper/Data.php [hunk 2]
```diff
@@ -1188,58 +1188,79 @@ public function getOpenInvoiceLineData(
...
+     * If it is not configured for the specific payment method,
+     * return the merchant account name defined in required settings.
+     *
+     * @param $paymentMethod
+     * @param int|null $storeId
+     * @return string
+     */
+    public function getAdyenMerchantAccount($paymentMethod, $storeId = null)
     {
         $merchantAccount = $this->getAdyenAbstractConfigData("merchant_account", $storeId);
```
Comment (rikterbeek) [inline Helper/Data.php]: as storeId can be null should we not retrieve it here or not make sure storeId needs to have an value ?

File: Model/AdyenInitiateTerminalApi.php [hunk 3]
```diff
@@ -37,43 +37,52 @@ class AdyenInitiateTerminalApi implements AdyenInitiateTerminalApiInterface
...
         \Magento\Checkout\Model\Session $_checkoutSession,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
         array $data = []
     )
     {
         $this->_adyenHelper = $adyenHelper;
         $this->_adyenLogger = $adyenLogger;
         $this->_checkoutSession = $_checkoutSession;
+        $this->storeManager = $storeManager;
+        $this->storeId = $this->storeManager->getStore()->getId();
```
Comment (rikterbeek) [inline Model/AdyenInitiateTerminalApi.php]: We can simplify this with
```
this->storeId = $storeManager->getStore()->getId();
```
so we do not have to expose storeManager variable.

File: Helper/Data.php (hunk 2 above)
Comment (Aleffio) [reply Helper/Data.php]: I will revert it to mandatory, as in all the places where we call it, it is specified

**Pull request merged**